docs/scale/04-deploy-app.md

```diff
@@ -0,0 +1,300 @@
+
+3. Change directory into the `swarm-microservice-demo-v1/web-vote-app` directory.
+
+        $ cd swarm-microservice-demo-v1/web-vote-app/
+
+4. View the Dockerfile` contents.
+
+        $ cat Dockerfile  
+        # Using official python runtime base image
+        FROM python:2.7
+        # Set the application directory
+        WORKDIR /app
+        # Install our requirements.txt
+        ADD requirements.txt /app/requirements.txt
+        RUN pip install -r requirements.txt
+        # Copy our code from the current folder to /app inside the container
+        ADD . /app
+        # Make port 80 available for links and/or publish
+        EXPOSE 80
+        # Define our command to be run when launching the container
+        CMD ["python", "app.py"]
+
+    As you can see, the image is based on the official `Python:2.7` tagged
+    image, adds a requirements file into the `/app` directory, installs
+    requirements, copies files from the build context into the container,
+    exposes port `80` and tells the container which command to run.
+
+5. Spend time investigating the other parts of the application by viewing the `results-app/Dockefile` and the `vote-worker/Dockerfile` in the application.
+
+## Step 2. Build custom images
+
+1. If you haven't already, `ssh` into the Swarm `manager` node.
+
+2. Make sure you have DOCKER_HOST set
+
+        $ export DOCKER_HOST="tcp://192.168.33.11:3375"
+
+3. Change to the root of your `swarm-microservice-demo-v1` clone.
+
+4. Build the `web-votes-app` image both the front end nodes.
+
+    **frontend01**:
+
+        $ docker -H tcp://192.168.33.20:2375 build -t web-vote-app ./web-vote-app
+
+    **frontend02**:
+
+        $ docker -H tcp://192.168.33.21:2375 build -t web-vote-app ./web-vote-app
+
+    These commands build the  `web-vote-app` image on the `frontend01` and
+    `frontend02` nodes. To accomplish the operation, each command copies the
+    contents of the `swarm-microservice-demo-v1/web-vote-app` sub-directory from the
+    `manager` node to each frontend node. The command then instructs the
+    Docker daemon on each frontend node to build the image and store it locally.
+
+    It may take a minute or so for each image to build. Wait for the builds to finish.
+
+5. Build `vote-worker` image on the `worker01` node
+
+        $ docker -H tcp://192.168.33.200:2375 build -t vote-worker ./vote-worker
+
+    It may take a minute or so for the image to build. Wait for the build to
+    finish.
+
+5. Build the `results-app` on the `store` node
+
+        $ docker -H tcp://192.168.33.250:2375 build -t results-app ./results-app
+
+Each of the *custom images* required by the application is now built and stored
+locally on the nodes that will use them.
+
+## Step 3. Pull images from Docker Hub
+
+For performance reasons, it is always better to pull any required Docker Hub
+images locally on each instance that needs them. This ensures that containers
+based on those images can start quickly.
```

Comment:
Actually, I think I'll add a note about this.  Conceptually, it is handy at this point to have the user specify the node because it helps them think about the architecture.